File: Models/Nrwdma.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class Nrwdma
{
    public int Id { get; set; }

    public int? MaDma { get; set; }

    public int Ky { get; set; }

    public int Nam { get; set; }

    public double LuongNuocVao { get; set; }

    public double LuongNuocTieuThu { get; set; }

    public double? LuongNuocSucXa { get; set; }

    public double? LuongNuocThatThoat { get; set; }

    public double? TyLeThatThoatKyTruoc { get; set; }

    public double? TyLeThatThoat { get; set; }

    public string? NguyenNhan { get; set; }

    public string? GhiChu { get; set; }

    public DateTime? NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual Dsdma? MaDmaNavigation { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;

    public virtual ICollection<NrwdmadauVaoChiTiet> NrwdmadauVaoChiTiets { get; set; } = new List<NrwdmadauVaoChiTiet>();

    public virtual ICollection<NrwdmatieuThuChiTiet> NrwdmatieuThuChiTiets { get; set; } = new List<NrwdmatieuThuChiTiet>();
}

File: src/types/nrw-dma/nrw-dma.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddNrwDmaRequest {
  MaDma?: number;
  Ky: number;
  Nam: number;
  LuongNuocVao: number;
  LuongNuocTieuThu: number;
  LuongNuocSucXa?: number;
  LuongNuocThatThoat?: number;
  TyLeThatThoatKyTruoc?: number;
  TyLeThatThoat?: number;
  NguyenNhan?: string;
  GhiChu?: string;
  NgayTao?: string;
  NguoiTao: number;
}

// Cập nhật dữ liệu
export interface UpdateNrwDmaRequest {
  MaDma?: number;
  Ky: number;
  Nam: number;
  LuongNuocVao: number;
  LuongNuocTieuThu: number;
  LuongNuocSucXa?: number;
  LuongNuocThatThoat?: number;
  TyLeThatThoatKyTruoc?: number;
  TyLeThatThoat?: number;
  NguyenNhan?: string;
  GhiChu?: string;
  NgayCapNhat?: string;
  NguoiCapNhat?: number;
}

/* ========== RESPONE (API --> CLIENT) ========== */
// Dữ liệu trả về
export interface NrwDmaResponse {
  id: number;
  maDma?: string;
  ky: number;
  nam: number;
  luongNuocVao: number;
  luongNuocTieuThu: number;
  luongNuocSucXa?: number;
  luongNuocThatThoat?: number;
  tyLeThatThoatKyTruoc?: number;
  tyLeThatThoat?: number;
  nguyenNhan?: string;
  ghiChu?: string;
  ngayTao?: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}